eva-accession-release-automation/gather_clustering_counts/gather_release_counts.py

```diff
@@ -228,6 +195,17 @@ def get_taxonomy_and_scientific_name(self, species_folder):
                 )
                 with get_metadata_connection_handle(self.config_profile, self.private_config_xml_file) as db_conn:
                     results = get_all_results_for_query(db_conn, query)
+        if len(results) < 1:
+            # Support for directory from release 1
+            if species_folder.split('_')[-1].isdigit():
+                taxonomy = int(species_folder.split('_')[-1])
+                query = (
+                    f"select distinct taxonomy_id, scientific_name "
+                    f"from evapro.taxonomy  "
+                    f"where taxonomy_id={taxonomy}"
+                )
+                with get_metadata_connection_handle(self.config_profile, self.private_config_xml_file) as db_conn:
+                    results = get_all_results_for_query(db_conn, query)
```

Comment:
I think your understanding of the code is correct. These different options came about organically to support loading old release data so some part will only be relevant to this loading exercise. 
I think for new run only the first one should be used i.e. we have the folder name of the release which we match against the tracking table to get the taxonomy which we match agains the metadata to get the scientific name.
The second option is only needed because we did not have all the taxonomies in EVAPRO. 
The third and forth options are only useful because release 1 is not in the tracker but their release folder does contain the taxonomy id. I agree that 4 could be before 3 but it doesn't really mater since all but one should be removed.

Lastly the scientific_name is not used here anymore so this whole function could be greatly simplified.